test/JoeLensV2.js

```diff
@@ -0,0 +1,260 @@
+      .connect(this.admin)
+      ._setRewardDistributor(this.rewardDistributor.address);
+
+    // Deploy Reward Lens for old joe lens and set rewards 
+    this.rewardLens = await this.RewardLensCF.deploy();
+    await this.rewardLens.deployed(); 
+
+    await this.rewardLens.setMarketRewards(
+      this.jUsdcNative.address, 
+      ethers.utils.parseEther("0.1"), 
+      ethers.utils.parseEther("0.3"), 
+      ethers.utils.parseEther("0.1"), 
+      ethers.utils.parseEther("0.3")
+    );
+
+    // Deploy JoeLens and JoeLensV2 
+    this.joeLensOld = await this.JoeLensCFOld.deploy(this.jUsdcNative, this.rewardLens.address);
+    this.joeLensNew = await this.JoeLensCFNew.deploy(this.jUsdcNative, this.rewardDistributor.address);
+    
+    await this.joeLensNew.setAdmin(this.dev.address);
+
+    // Deploy price oracle and set aggregators
+    this.priceOracle = await this.PriceOracleCF.deploy(this.dev.address);
+    await this.priceOracle.deployed();
+
+    await this.priceOracle
+      .connect(this.dev)._setAggregators(
+        [this.jUsdcNative.address],
+        [JUSDC_NATIVE_AGGREGATOR_ADDRESS]
+      );
+
+    await this.joetroller
+      .connect(this.admin)
+      ._setPriceOracle(this.priceOracle.address);
+
+    this.usdcToken = await ethers.getContractAt("EIP20Interface", USDC_ADDRESS);
+    // Approve the spender
+    await this.usdcToken
+      .connect(this.dev)
+      .approve(this.jUsdcNative.address, ethers.utils.parseEther("1000"));
+
+    // mint jusdc
+    await this.jUsdcNative
+      .connect(this.dev)
+      .mint(ethers.utils.parseUnits("1000", 6));
+  });
+
+  it("Data returned from jTokenMetadata() is the same as it was from the old JoeLens", async function () {
+    const oldMarketData = await this.joeLensOld.callStatic.jTokenMetadata(this.jUsdcNative.address);
+    const newMarketData = await this.joeLensNew.callStatic.jTokenMetadata(this.jUsdcNative.address);
+
+    // Minimize returned data to only include fields in the jTokenMetadata struct
+    const parsedOldMarketData = Object.fromEntries(
+      Object.entries(oldMarketData).slice(26, 44)
+    );
+    const parsedNewMarketData = Object.fromEntries(
+      Object.entries(newMarketData).slice(26, 44)
+    );
+
+    expect(JSON.stringify(parsedOldMarketData)).to.be.equal(JSON.stringify(parsedNewMarketData));
+  });
+
+  it("Data returned from jTokenMetadataAll() is the as it was from the old JoeLens", async function () {
+    const oldMarketData = await this.joeLensOld.callStatic.jTokenMetadataAll([this.jUsdcNative.address]);
+    const newMarketData = await this.joeLensNew.callStatic.jTokenMetadataAll([this.jUsdcNative.address]);
+
+    // Minimize returned data to only include fields in the JTokenMetadata struct
+    const parsedOldMarketData = Object.fromEntries(
+      Object.entries(oldMarketData[0]).slice(26, 44)
+    );
+    const parsedNewMarketData = Object.fromEntries(
+      Object.entries(newMarketData[0]).slice(26, 44)
+    );
```

Comment:
We remove the last 6 fields? 

Last 4 I could understand - because the reward speeds were different; although I suggest we don't change reward speed on RewardLens.

The other two however - supplyCap and borrowCap - these should be the same.

In fact I would not even both slicing the objects and just directly compare they're exactly the same.

Reply:
Yea makes more sense to just compare them directly in hindsight, this is overcomplicated